xarray/core/indexes.py

```diff
@@ -674,10 +670,11 @@ def _concat_indexes(indexes, dim, positions=None) -> pd.Index:
     @classmethod
     def concat(
         cls,
-        indexes: Sequence[PandasIndex],
+        # Currently only compatible with PandasIndex, which fails the Liskov substitution principle
+        indexes: Sequence[Self],  # type: ignore[override]
```

Comment:
If you change the type of indexes to `Sequence[Self]` in `Index.concat` as well, does this work?
Or is it supposed to work with any indexes in the future?

Reply:
Very good idea! This removes lots of ignores